atomic_reactor/util.py

```diff
@@ -852,4 +853,226 @@ def df_parser(df_path, workflow=None, cache_content=False, env_replace=True, par
+        self.LABEL_TYPE_COMPONENT = 7
+        self.LABEL_TYPE_HOST = 8
+
+        self._label_names = {
+            self.LABEL_TYPE_NAME : ('name', 'Name'),
+            self.LABEL_TYPE_VERSION : ('version', 'Version'),
+            self.LABEL_TYPE_RELEASE : ('release', 'Release'),
+            self.LABEL_TYPE_ARCH : ('architecture', 'Architecture'),
+            self.LABEL_TYPE_VENDOR : ('vendor', 'Vendor'),
+            self.LABEL_TYPE_SOURCE : ('authoritative-source-url', 'Authoritative_Registry'),
+            self.LABEL_TYPE_COMPONENT : ('com.redhat.component', 'BZComponent'),
+            self.LABEL_TYPE_HOST : ('com.redhat.build-host', 'Build_Host')
+        }
+
+        self._df_labels = df_labels
+        self._label_values = {}
+        for label_type in self._label_names:
+            for lbl_name in self._label_names[label_type]:
+                if lbl_name in df_labels:
+                    self._label_values[label_type] = df_labels[lbl_name]
+                    break
+
+    def get_label_only(self, label_type):
+        """returns newest label name"""
+        return self._label_names[label_type][0]
+
+    def get_label_name(self):
+        return self.get_label_only(self.LABEL_TYPE_NAME)
+
+    def get_label_version(self):
+        return self.get_label_only(self.LABEL_TYPE_VERSION)
+
+    def get_label_release(self):
+        return self.get_label_only(self.LABEL_TYPE_RELEASE)
+
+    def get_label_arch(self):
+        return self.get_label_only(self.LABEL_TYPE_ARCH)
+
+    def get_label_vendor(self):
+        return self.get_label_only(self.LABEL_TYPE_VENDOR)
+
+    def get_label_source(self):
+        return self.get_label_only(self.LABEL_TYPE_SOURCE)
+
+    def get_label_component(self):
+        return self.get_label_only(self.LABEL_TYPE_COMPONENT)
+
+    def get_label_host(self):
+        return self.get_label_only(self.LABEL_TYPE_HOST)
+
+    def get_old_labels(self, label_type):
+        """returns tuple with all old label names"""
+        return self._label_names[label_type][1:]
+
+    def get_old_labels_name(self):
+        return self.get_old_labels(self.LABEL_TYPE_NAME)
+
+    def get_old_labels_version(self):
+        return self.get_old_labels(self.LABEL_TYPE_VERSION)
+
+    def get_old_labels_release(self):
+        return self.get_old_labels(self.LABEL_TYPE_RELEASE)
+
+    def get_old_labels_arch(self):
+        return self.get_old_labels(self.LABEL_TYPE_ARCH)
+
+    def get_old_labels_vendor(self):
+        return self.get_old_labels(self.LABEL_TYPE_VENDOR)
+
+    def get_old_labels_source(self):
+        return self.get_old_labels(self.LABEL_TYPE_SOURCE)
+
+    def get_old_labels_component(self):
+        return self.get_old_labels(self.LABEL_TYPE_COMPONENT)
+
+    def get_old_labels_host(self):
+        return self.get_old_labels(self.LABEL_TYPE_HOST)
+
+    def get_all_labels(self, label_type):
+        """returns tuple with all label names"""
+        return self._label_names[label_type]
```

Comment:
That's what `get_new_names_by_old()` is for isn't it? For the add_labels_in_dockerfile plugin to use because it needs to know the complete mapping.